--- /usr/bin/mysqld_safe.orig	2014-07-06 23:05:31.378738591 -0700
+++ /usr/bin/mysqld_safe	2014-07-06 23:05:12.158643283 -0700
@@ -143,7 +143,7 @@
 eval_log_error () {
   cmd="$1"
   case $logging in
-    file) cmd="$cmd 2>&1 | "`shell_quote_string "$helper"`" $user log "`shell_quote_string "$err_log"` ;;
+    file) cmd="$cmd 2>&1 | "`shell_quote_string "$helper"`" $user log "`shell_quote_string "$err_log"`" & wait" ;;
     syslog)
       # mysqld often prefixes its messages with a timestamp, which is
       # redundant when logging to syslog (which adds its own timestamp)
